src/System.Collections.NonGeneric/src/System/Collections/CaseInsensitiveHashCodeProvider.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+
+namespace System.Collections
+{
+    /// <summary>
+    /// Designed to support hashtables which require case-insensitive behavior while still maintaining case,
+    /// this provides an efficient mechanism for getting the hashcode of the string ignoring case.
+    /// </summary>
+    [Obsolete("Please use StringComparer instead.")]
+    public class CaseInsensitiveHashCodeProvider : IHashCodeProvider
+    {
+        private static volatile CaseInsensitiveHashCodeProvider s_invariantCaseInsensitiveHashCodeProvider = null;
+        private readonly StringComparer _comparer;
+
+        public CaseInsensitiveHashCodeProvider() : 
+            this(StringComparer.CurrentCultureIgnoreCase)
+        {
+        }
+
+        public CaseInsensitiveHashCodeProvider(CultureInfo culture)
+        {
+            if (culture == null)
+            {
+                throw new ArgumentNullException(nameof(culture));
+            }
+
+            //_comparer = StringComparer.Create(culture, ignoreCase: true); // TODO #9664: This is broken until Create is exposed
+            _comparer = StringComparer.CurrentCultureIgnoreCase;
```

Comment:
Yeah.  But I can't think of anything better... you?  All of the required support appears to be internal to mscorlib until StringComparer.Create is exposed.